pkg/datastore/version.go

```diff
@@ -0,0 +1,64 @@
+package datastore
+
+import (
+	"fmt"
+	"github.com/coreos/go-semver/semver"
+	"github.com/interuss/stacktrace"
+	"regexp"
+)
+
+type VersionRegex struct {
+	Name         string
+	VersionRegex string
+}
+
+type Version struct {
+	version *semver.Version
+	dsName  string
+}
+
+var COCKROACHDB = &VersionRegex{"cockroach", `v((0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*))`}
+var YUGABYTE = &VersionRegex{"yugabyte", `-YB-((0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*))`}
+
+func (dt *VersionRegex) ParseVersion(fullVersion string) (*semver.Version, error) {
+	re := regexp.MustCompile(dt.VersionRegex)
```

Comment:
Store this as a variable instead of the regex string: since it is hardcoded better panic early rather than when attempting to use it.